From: Matthew Wilcox <matthew.r.wilcox@intel.com>
Date: Fri, 9 May 2014 22:42:26 -0400
Subject: NVMe: Enable BUILD_BUG_ON checks
References: bnc#913030,FATE#317455
Patch-Mainline: v3.17
Git-commit: 21bd78bcf4208e84deab0d34f9d4e034d0580d0c

Since _nvme_check_size() wasn't being called from anywhere, the compiler
was optimising it away ... along with all the link-time build failures
that would result if any of the structures were the wrong size.  Call it
from nvme_exit() for no particular reason.

Signed-off-by: Matthew Wilcox <matthew.r.wilcox@intel.com>
Acked-by: Hannes Reinecke <hare@suse.de>
---
 drivers/block/nvme-core.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/block/nvme-core.c b/drivers/block/nvme-core.c
index 76f3dff..1c0cbc9 100644
--- a/drivers/block/nvme-core.c
+++ b/drivers/block/nvme-core.c
@@ -2997,6 +2997,7 @@ static void __exit nvme_exit(void)
 	unregister_blkdev(nvme_major, "nvme");
 	destroy_workqueue(nvme_workq);
 	BUG_ON(nvme_thread && !IS_ERR(nvme_thread));
+	_nvme_check_size();
 }
 
 MODULE_AUTHOR("Matthew Wilcox <willy@linux.intel.com>");
-- 
1.8.5.2

